src/main/java/org/veupathdb/service/eda/compute/plugins/differentialabundance/DifferentialAbundancePlugin.java

```diff
@@ -0,0 +1,130 @@
+      connection.voidEval("absoluteAbundanceData <- " + INPUT_DATA); // Need to rename so we can go get the sampleMetadata later
+
+      // Read in the sample metadata
+      List<VariableSpec> sampleMetadataVars = ListBuilder.asList(comparisonVariableSpec);
+      sampleMetadataVars.add(computeEntityIdVarSpec);
+      System.out.println(sampleMetadataVars);
+      connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, sampleMetadataVars));
+      connection.voidEval("sampleMetadata <- " + INPUT_DATA);
+
+
+      // TODO make a helper for this i think
+      List<String> dotNotatedIdColumns = idColumns.stream().map(VariableDef::toDotNotation).collect(Collectors.toList());
+      String dotNotatedIdColumnsString = "c(";
+      boolean first = true;
+      for (String idCol : dotNotatedIdColumns) {
+        if (first) {
+          first = false;
+          dotNotatedIdColumnsString = dotNotatedIdColumnsString + util.singleQuote(idCol);
+        } else {
+          dotNotatedIdColumnsString = dotNotatedIdColumnsString + "," + util.singleQuote(idCol);
+        }
+      }
+      dotNotatedIdColumnsString = dotNotatedIdColumnsString + ")";
+
+      // TEMP FOR TESTING ONLY - REMOVE WHEN ABSOLUTE ABUNDANCES ARE HERE
+      connection.voidEval("taxaColNames <- names(absoluteAbundanceData[, -c('" + computeEntityIdColName + "', as.character(" + dotNotatedIdColumnsString + "))])");
+      connection.voidEval("absoluteAbundanceData[, (taxaColNames) := lapply(.SD,function(x) {round(x*1000)}), .SDcols=taxaColNames]");
+      // END OF TEMP FOR TESTING
+
+      connection.voidEval("diffabundDT <- microbiomeComputations::AbsoluteAbundanceData(data=absoluteAbundanceData" + 
+                                                                          ", sampleMetadata=sampleMetadata" +
+                                                                          ", recordIdColumn=" + PluginUtil.singleQuote(computeEntityIdColName) + 
+                                                                          ", ancestorIdColumns=as.character(" + dotNotatedIdColumnsString + ")" +
+                                                                          ", imputeZero=TRUE)");
+
+
+      connection.voidEval("differentialabundanceDT <- differentialAbundance(data=diffabundDT" +
+                                                          ", comparisonVariable=" + PluginUtil.singleQuote(comparisonVariable) +
+                                                          ", groupA=" + groupA +
+                                                          ", groupB=" + groupB + 
+                                                          ", method=" + PluginUtil.singleQuote(method) + 
+                                                          ", verbose=TRUE)");
+
+
+
+      String dataCmd = "writeData(differentialabundanceDT, NULL, TRUE)";
+      String metaCmd = "writeMeta(differentialabundanceDT, NULL, TRUE)";
```

Comment:
if we dont mean to use the data and meta endpoints, do we need to write them? im not sure..

Reply:
No, you do not! :)  If you don't, then requests to those file endpoints should return 404, which I think is what we want.